leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/HeapAnalysisListScreen.kt

```diff
@@ -70,10 +65,17 @@ internal class HeapAnalysisListScreen : Screen() {
               view.context, projection.createdAtTimeMillis,
               DateUtils.FORMAT_SHOW_TIME or DateUtils.FORMAT_SHOW_DATE
           )
-
-          titleView.text = "$index. " + (projection.exceptionSummary
-          // TODO Handle singular
-              ?: "${projection.retainedInstanceCount} retained instances")
+          when {
+            projection.exceptionSummary != null -> titleView.text =
+              resources.getString(
+                  R.string.leak_canary_heap_analysis_list_exception_summary, index,
+                  projection.exceptionSummary
+              )
+            else -> titleView.text = resources.getQuantityText(
```

Comment:
1) move `titleView.text =` out of when?
2) nit: check retainedInstanceCount vs using `resource.getQuantityText(...)`?

Reply:
Moved the assignment out of the `when`. What's the reasoning for not using `getQuantityText`?